src/dotenv/parser.py

```diff
@@ -172,9 +172,9 @@ def parse_unquoted_value(reader):
         (part,) = reader.read_regex(_unquoted_value_part)
         value += part
         after = reader.peek(2)
-        if len(after) < 2 or after[0] in u"\r\n" or after[1] in u" #\r\n":
-            return value
-        value += reader.read(2)
+        if after[:1] not in (u" ", u"\t") or after in (u" #", u"\t#"):
```

Comment:
Well, since `_unquoted_value_part` is `\S*`, and `\S` is not just `[^ \t]`, I could give you the following case:

```py
    (
        u"a=b\u00a0 c",
        [Binding(key=u"a", value=u"b\u00a0 c", original=Original(string=u"a=b\u00a0 c", line=1), error=False)],
    ),
```

But that's not the point. Isn't the way to parse the unquoted value in and of itself overcomplicated? With this the tests pass:

```py
def parse_unquoted_value(reader):
    # type: (Reader) -> Text
    (part,) = reader.read_regex(make_regex(r'([^\r\n]*)'))
    return re.sub(r'\s+#.*', '', part).rstrip()
```

Reply:
Excellent suggestion, thanks!  I've updated the code accordingly.